### PR TITLE
Blacklist a number of prediction field names.

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR prevents field name clashes in ml results docs on JSON level.

Relates https://github.com/elastic/elasticsearch/issues/48808

### CODE REVIEW

File: lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc [hunk 1]
```diff
@@ -68,6 +68,13 @@ CDataFrameTrainBoostedTreeClassifierRunner::CDataFrameTrainBoostedTreeClassifier
                   this->dependentVariableFieldName()) == categoricalFieldNames.end()) {
         HANDLE_FATAL(<< "Input error: trying to perform classification with numeric target.");
     }
+    const std::set<std::string> predictionFieldNameBlacklist{
+        IS_TRAINING_FIELD_NAME, PREDICTION_PROBABILITY_FIELD_NAME, TOP_CLASSES_FIELD_NAME};
+    if (predictionFieldNameBlacklist.count(this->predictionFieldName())) {
```
Comment (edsavage) [inline lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc]: I'd prefer an explicit comparison here, e.g `if (predictionFieldNameBlacklist.count(this->predictionFieldName()) > 0)`
Looking through our code there's no consistency but the bulk of such comparisons are explicit.
Incidentally there's nothing about this in the `STYLEGUIDE` either, maybe there should be?

File: lib/api/CDataFrameTrainBoostedTreeRegressionRunner.cc [hunk 2]
```diff
@@ -52,6 +52,12 @@ CDataFrameTrainBoostedTreeRegressionRunner::CDataFrameTrainBoostedTreeRegression
                   this->dependentVariableFieldName()) != categoricalFieldNames.end()) {
         HANDLE_FATAL(<< "Input error: trying to perform regression with categorical target.");
     }
+    const std::set<std::string> predictionFieldNameBlacklist{IS_TRAINING_FIELD_NAME};
+    if (predictionFieldNameBlacklist.count(this->predictionFieldName())) {
```
Comment (edsavage) [inline lib/api/CDataFrameTrainBoostedTreeRegressionRunner.cc]: Same comment as above applies here...

File: lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc (hunk 1 above)
Comment (droberts195) [reply lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc]: We used to have this in the old Word doc coding standards - section 5.27: "if statements SHOULD be fully qualified".  It got lost when that was converted to markdown.

Comment (edsavage) [reply lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc]: Created #862 to add this to the `STYLEGUIDE`

Comment (przemekwitek) [reply lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc]: > I'd prefer an explicit comparison here, e.g `if (predictionFieldNameBlacklist.count(this->predictionFieldName()) > 0)`

Done.

File: lib/api/CDataFrameTrainBoostedTreeRegressionRunner.cc (hunk 2 above)
Comment (przemekwitek) [reply lib/api/CDataFrameTrainBoostedTreeRegressionRunner.cc]: Done.

**Pull request merged**